src/ast/mod.rs

```diff
@@ -1107,14 +1123,31 @@ pub enum Action {
     // If(Expr, Action, Action),
 }
 
+#[derive(Clone, Debug, Copy, PartialEq, Eq, Hash)]
+pub enum ChangeRow {
+    Delete,
+    Subsume,
+    Unextractable,
+}
+
+impl ChangeRow {
+    pub fn to_action(&self, op: Symbol, args: Vec<Expr>) -> Action {
+        match self {
+            ChangeRow::Delete => Action::Delete(op, args),
+            ChangeRow::Subsume => Action::Subsume(op, args),
+            ChangeRow::Unextractable => Action::Unextractable(op, args),
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub enum NormAction {
     Let(Symbol, NormExpr),
     LetVar(Symbol, Symbol),
     LetLit(Symbol, Literal),
     Extract(Symbol, Symbol),
     Set(NormExpr, Symbol),
-    Delete(NormExpr),
```

Comment:
I collapsed this into one normalized action, since much of the behavior is the same for all three actions.

Reply:
great!